### PR TITLE
Move to Clarifai-GRPC library

### LINKED ISSUE #469: clarifai deprecation ; clarifai-grpc support ?
Hi everyone,

Thanks for making such a great package ! 

I'm running pliers on a series of videos, and I'd like to include the `ClarifaiAPIImageExtractor` transformer. I get a `KeyError`, though, when running the transformer if the python package `clarifai` is in my conda environment. 

Looking at the clarifai docs, I realized that this package has actually been deprecated in favor of [clarifai-grpc](https://github.com/Clarifai/clarifai-python-grpc). It seems from the [dependencies](https://github.com/PsychoinformaticsLab/pliers/blob/32e41e1aa3274faf1dcc2a3a70a90f35fd17ae1c/optional-dependencies.txt#L1), though, that pliers is still using the original, now deprecated version of the clarifai API.

Is this something that could be updated ? 

`pliers` version : 0.4.1
full `pip freeze` (with the current `clarifai-grpc`) :

<details>

```
asttokens @ file:///opt/conda/conda-bld/asttokens_1646925590279/work
backcall @ file:///home/ktietz/src/ci/backcall_1611930011877/work
cachetools==5.2.0
certifi @ file:///opt/conda/conda-bld/certifi_1655968806487/work/certifi
charset-normalizer==2.1.1
clarifai-grpc==8.8.0
click==8.1.3
configparser==3.8.1
decorator==4.4.2
executing @ file:///opt/conda/conda-bld/executing_1646925071911/work
face-recognition-models==0.3.0
flatbuffers==1.12
future==0.18.2
google-api-core==2.8.2
google-api-python-client==2.58.0
google-auth==2.11.0
google-auth-httplib2==0.1.0
googleapis-common-protos==1.56.4
grpcio==1.47.0
httplib2==0.20.4
idna==3.3
imageio==2.21.2
imageio-ffmpeg==0.4.7
ipython @ file:///opt/conda/conda-bld/ipython_1657652213665/work
jedi @ file:///tmp/build/80754af9/jedi_1644297102865/work
joblib==1.1.0
jsonschema==2.6.0
keras==2.9.0
libclang==14.0.1
matplotlib-inline @ file:///tmp/build/80754af9/matplotlib-inline_1628242447089/work
moviepy==1.0.3
murmurhash==1.0.7
nltk==3.7
numpy==1.23.2
pandas==1.4.3
parso @ file:///opt/conda/conda-bld/parso_1641458642106/work
pexpect @ file:///tmp/build/80754af9/pexpect_1605563209008/work
pickleshare @ file:///tmp/build/80754af9/pickleshare_1606932040724/work
Pillow==9.2.0
pliers==0.4.1
proglog==0.1.10
prompt-toolkit @ file:///tmp/build/80754af9/prompt-toolkit_1633440160888/work
protobuf==3.19.3
ptyprocess @ file:///tmp/build/80754af9/ptyprocess_1609355006118/work/dist/ptyprocess-0.7.0-py2.py3-none-any.whl
pure-eval @ file:///opt/conda/conda-bld/pure_eval_1646925070566/work
pyasn1==0.4.8
pyasn1-modules==0.2.8
Pygments @ file:///opt/conda/conda-bld/pygments_1644249106324/work
pyparsing==3.0.9
python-dateutil==2.8.2
python-magic==0.4.27
pytz==2022.1
regex==2022.8.17
requests==2.28.1
rsa==4.9
scipy==1.9.1
six @ file:///tmp/build/80754af9/six_1644875935023/work
SpeechRecognition==3.8.1
stack-data @ file:///opt/conda/conda-bld/stack_data_1646927590127/work
tensorboard-plugin-wit==1.8.1
termcolor==1.1.0
tokenizers==0.12.1
tqdm==4.64.0
traitlets @ file:///tmp/build/80754af9/traitlets_1636710298902/work
uritemplate==4.1.1
urllib3==1.26.12
wasabi==0.9.1
wcwidth @ file:///Users/ktietz/demo/mc3/conda-bld/wcwidth_1629357192024/work
```

</details>

### PR DESCRIPTION
**Pull request by adelavega**
Closes #469 

In order to make a call now, you also need to provide a `CLARIFAI_USER_ID` and `CLARIFAI_APPLICATION_ID` in addition to your `CLARIFAI_ACCESS_TOKEN`.

TODO:

- [x] Add ability to filter concepts
- [x] Update docs (re: authentication)
- [x] Ingest clarifai responses
- [x] Check tests

### CODE REVIEW

**Comment by adelavega** [comment]
Example of what I'm getting for `obama.jpg`:

image

**Comment by adelavega** [comment]
@emdupre Tests are passing with the new library.

As far as the user is concerned, there shouldn't be too many differences except that now you need: A `CLARIFAI_USER_ID` and `CLARIFAI_APPLICATION_ID` in addition to your `CLARIFAI_ACCESS_TOKEN`. 

When you sign up for Clarifai, you can get your PAT from your profile, and you create an app which will then generate the other two IDs.

Tests are probably going to fail in CI but they are passing locally, so give it a shot in this PR if you have time. 

Otherwise, this is ready to merge. Unless @qmac you want to review real quick.

**Comment by emdupre** [comment]
Thanks, @adelavega  !! I have it annotating some stimuli now, so these changes definitely get things running again ! 🚀 

A few small things I was confused by :

- The environmental variable needs to be named `CLARIFAI_APP_ID` rather than `CLARIFAI_APPLICATION_ID`
- The Transformer is still  `ClarifaiAPIImageExtractor` rather than the more generic `ClarifaiAPIExtractor` that I interpreted from [this change](https://github.com/PsychoinformaticsLab/pliers/pull/474/files#diff-d9b149498982c0663c3b7170398773361ed5678f1a627e9c2fd8d2c955c563dbL117-R119).

**Comment by adelavega** [comment]
Great!

Thanks for that catch on the environment name.

In the docs when I said "(and subclasses) " I meant to include the ImageExtractor.

Maybe I could put: "e.g. `ClarifaiImageExtractor`" to clarify.

File: docs/installation.rst [hunk 1]
```diff
@@ -114,9 +114,9 @@ While installing pliers itself is usually straightforward, setting up some of th
...
-| ClarifaiAPIVideoExtractor             | `Clarifai video tagging API <https://clarifai.com>`__                                               | CLARIFAI\_API\_KEY                   | API key                        | 168ed02e137459ead66c3a661be7b784      |
+| ClarifaiAPIExtractor                  | `Clarifai recognition API <https://clarifai.com>`__                                                 | CLARIFAI\_ACCESS_TOKEN               | Personal access token          | 168ed02e137459ead66c3a661be7b784      |
+| (and subclasses                       |                                                                                                     | CLARIFAI\_USER\_ID                   | User name                      | user_name                             |
+|  e.g ClarifaiImageExtractor)          |                                                                                                     | CLARIFAI\_APP\_ID            | Application name                       | my_application                        |
```
Comment (emdupre) [inline docs/installation.rst:119]:
```suggestion
|  e.g  ClarifaiAPIImageExtractor)          |                                                                                                     | CLARIFAI\_APP\_ID            | Application name                       | my_application                        |
```

**Review by emdupre — APPROVED** [review]
This LGTM ! Thank you again for working on this 🙏

**Comment by adelavega** [comment]
Awesome thank you! 

Tests are passing locally for the relevant tests, so I'm merging

**Pull request merged**